exporter/collector/googlemanagedprometheus/monitoredresource.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlemanagedprometheus
+
+import (
+	"go.opentelemetry.io/collector/model/pdata"
+	semconv "go.opentelemetry.io/collector/model/semconv/v1.8.0"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+)
+
+func MapToPrometheusTarget(res pdata.Resource) *monitoredrespb.MonitoredResource {
+	attrs := res.Attributes()
+	// Prepend namespace if it exists to match what is specified in
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#resource-attributes-1
+	job := getStringOrEmpty(attrs, semconv.AttributeServiceName)
+	serviceNamespace := getStringOrEmpty(attrs, semconv.AttributeServiceNamespace)
+	if serviceNamespace != "" {
+		job = serviceNamespace + "/" + job
+	}
+	return &monitoredrespb.MonitoredResource{
+		Type: "prometheus_target",
+		Labels: map[string]string{
+			"location":  getStringOrEmpty(attrs, semconv.AttributeCloudAvailabilityZone),
+			"cluster":   getStringOrEmpty(attrs, semconv.AttributeK8SClusterName),
+			"namespace": getStringOrEmpty(attrs, semconv.AttributeK8SNamespaceName),
+			"job":       job,
+			"instance":  getStringOrEmpty(attrs, semconv.AttributeServiceInstanceID),
```

Comment:
For Managed GMP, `instance` label is set to `<pod_name>:<port number>` or `<pod_name>:<port name>`. https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/pkg/operator/apis/monitoring/v1/types.go#L508

Self-managed GMP sets the `instance` label to the Prometheus `instance` label (https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/pkg/export/setup/setup.go#L70).

What would it look like for a process using GMP exporter on GKE or GCE?

Reply:
Right now, this sets `instance` to the prometheus `instance` label.  The prometheus receiver converts `instance` to `service.instance.id`, which is the OTel equivalent.

If you want, I can change this to prefer using `<pod_name>:<port number>`, or `<pod_name>:<port name>`if those attributes are available.  Just let me know which should take precedence, and i'll add it.

Reply:
I generally see the OT setup w/ Prometheus receiver and GMP exporter resembles the self-deployed GMP experience as opposed to the managed experience. If so, using whatever `instance` label is set by user's Prometheus configuration makes sense as is done here, but we should note this creates a different experience from managed GMP. cc/ @jsuereth

Reply:
The managed GMP link above just points to relabel rules added by the managed GMP operator.  You **can** still use relabel rules for `instance` in the otel Prometheus receiver (e.g. the way managed GMP does), and it will correctly change the instance label on the exporter.  So it should work in both cases.